core/retries-api/src/main/java/software/amazon/awssdk/retries/api/RetryStrategy.java

```diff
@@ -191,17 +197,12 @@ default Builder retryOnRootCauseInstanceOf(Class<? extends Throwable> throwable)
          *
          * <p>The default value for the standard and adaptive retry strategies is 3.
          */
-        Builder maxAttempts(int maxAttempts);
-
-        /**
-         * Configure the predicate to allow the strategy categorize a Throwable as throttling exception.
-         */
-        Builder treatAsThrottling(Predicate<Throwable> treatAsThrottling);
```

Comment:
This is only needed for `AdaptiveRetryStrategy` and `LegacyRetryStrategy`. I will move to the corresponding interfaces instead.